spdx-utils/src/main/resources/declared-license-mapping.yml

```diff
@@ -1,6 +1,7 @@
 # Sort the entries below via IntelliJ's "Edit" -> "Sort Lines".
 # Map a declared license string to "NONE" in order to discard it.
 ---
+"(MIT OR EUPL-1.1+)": MIT OR EUPL-1.1-or-later
```

Comment:
I don't see "EUPL-1.1-or-later" in the SPDX license list. Also, wouldn't the declared license processor handle this license already correctly?

Reply:
> I don't see "EUPL-1.1-or-later" in the SPDX license list.

Would you be fine with mapping it to just "MIT OR EUPL-1.1" ?

> Also, wouldn't the declared license processor handle this license already correctly?

It didn't.

Reply:
> Would you be fine with mapping it to just "MIT OR EUPL-1.1" ?

Shouldn't we then do "MIT OR EUPL-1.1 OR EUPL-1.2" to mimic what "or later" (currently) means?

Reply:
Done.

Reply:
The declared license processor fails, because `isValid()` returns `false` for "EUPL-1.1+". I believe this is a bug in `SpdxLicenseIdExpression`. Instead of `private val spdxLicense = SpdxLicense.forId(toString())` it should do `private val spdxLicense = SpdxLicense.forId(id)`, because otherwise all licenses with a `+` are seen as not valid.

https://github.com/oss-review-toolkit/ort/blob/c27b549f78976da479abc8e4e7001056c31e6879/spdx-utils/src/main/kotlin/SpdxExpression.kt#L404-L421

Reply:
Thanks @mnonnenmacher for debugging it. I've ditched this commit in favor of your proposed fix.